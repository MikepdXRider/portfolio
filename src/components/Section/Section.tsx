import React from 'react';
import styles from './section.module.css';
import '../../App.css';

interface SectionProps {
  children: React.ReactChild;
  title?: string;
  darkMode?: boolean;
}

export default function Section({
  children,
  title,
  darkMode = false,
}: SectionProps) {
  return (
    <section
      aria-label={`${title} section`}
      className={`${
        darkMode ? styles.darkModeSection : styles.lightModeSection
      } ${title ? styles.sectionWithTitle : styles.section}`}
      id={title}
    >
      {title && (
        <article className={styles.titleContainer}>
          <h2
            className={`${
              darkMode ? styles.darkModeTitle : styles.lightModeTitle
            } ${styles.title}`}
          >
            {title}
          </h2>
        </article>
      )}

      {children}
    </section>
  );
}
