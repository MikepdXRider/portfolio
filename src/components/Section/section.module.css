:root {
  /* alternatives: hsl(225,7%,22%) | rgb(53,55,61)*/
  --darkslategrey: #35373d;
  /* alternatives: hsl(0,100%,100%) | rgb(255,255,255)*/
  --white: #ffffff;
  /* alternatives: hsl(0,0%,100%) | rgb(0,0,0)*/
  --black: #000000;
  /* alternatives: hsl(90,58%,56%) | rgb(143,209,79)*/
  --accentgreen: #8fd14f;
}

.section {
  min-height: 500px;
  min-width: 100%;
  display: grid;
  grid-template-rows: 1fr 4fr;
}

.darkModeSection {
  background: var(--darkslategrey);
}

.lightModeSection {
  background: var(--white);
}

.title {
  grid-row: 0 / 1;
  text-decoration: underline;
  text-decoration-color: var(--accentgreen);
  -moz-text-decoration-color: var(--accentgreen);
  text-decoration-thickness: 5px;
}

.darkModeTitle {
  color: var(--white);
}

.lightModeTitle {
  color: var(--black);
}
