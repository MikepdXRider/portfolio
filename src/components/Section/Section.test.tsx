import { render, screen } from '@testing-library/react';
import Section from './Section';

describe('section component', () => {
  it('renders section component WITH a title and children', () => {
    render(
      <Section title={'test-title'}>
        <p>Hello World</p>
      </Section>
    );

    // Grabs elements from rendered DOM
    const titleEl = screen.getByText(/test-title/i);
    const childEl = screen.getByText(/hello world/i);

    // Checks that elements are in rendered DOM
    expect(titleEl).toBeInTheDocument();
    expect(childEl).toBeInTheDocument();
  });

  it('matches the existing snapshot WITH a title and children', () => {
    const { container } = render(
      <Section title={'test-title'}>
        <p>Hello World</p>
      </Section>
    );

    expect(container).toMatchSnapshot();
  });

  it('renders section component WITHOUT a title and WITH children', () => {
    render(
      <Section>
        <p>Hello World</p>
      </Section>
    );

    // Grabs elements from rendered DOM
    const childEl = screen.getByText(/hello world/i);

    // Checks that elements are in rendered DOM
    expect(childEl).toBeInTheDocument();
  });

  it('matches the existing snapshot WITHOUT a title and WITH children', () => {
    const { container } = render(
      <Section>
        <p>Hello World</p>
      </Section>
    );

    expect(container).toMatchSnapshot();
  });
});
