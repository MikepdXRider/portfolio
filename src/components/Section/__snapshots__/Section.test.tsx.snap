// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`section component matches the existing snapshot WITH a title and children 1`] = `
<div>
  <section>
    <title>
      test-title
    </title>
    <p>
      Hello World
    </p>
  </section>
</div>
`;

exports[`section component matches the existing snapshot WITHOUT a title and WITH children 1`] = `
<div>
  <section>
    <p>
      Hello World
    </p>
  </section>
</div>
`;
